sqlobject/src/main/java/org/jdbi/v3/sqlobject/SqlObjectFactory.java

```diff
@@ -87,182 +64,29 @@ private boolean looksLikeSqlObject(Class<?> extensionType) {
      * Create a sql object of the specified type bound to this handle. Any state changes to the handle, or the sql
      * object, such as transaction status, closing it, etc, will apply to both the object and the handle.
      *
-     * @param extensionType the type of sql object to create
+     * @param extensionType  the type of sql object to create
      * @param handleSupplier the Handle instance to attach ths sql object to
      * @return the new sql object bound to this handle
      */
     @Override
     public <E> E attach(Class<E> extensionType, HandleSupplier handleSupplier) {
-        final SqlObjectInitData sqlObjectInitData = sqlObjectCache.get(extensionType, handleSupplier.getConfig());
-        final ConfigRegistry instanceConfig = sqlObjectInitData.configureInstance(handleSupplier.getConfig().createCopy());
-
         if (SqlObjectInitData.isConcrete(extensionType)) {
-            return sqlObjectInitData.instantiate(extensionType, handleSupplier, instanceConfig);
+            throw new IllegalStateException(format("Can not process %s, it is a generated SQL object", extensionType.getSimpleName()));
         }
 
+        final SqlObjectInitData sqlObjectInitData = SqlObjectInitData.lookup(extensionType, handleSupplier.getConfig());
+        final ConfigRegistry instanceConfig = sqlObjectInitData.configureInstance(handleSupplier.getConfig().createCopy());
+
         instanceConfig.get(Extensions.class).onCreateProxy();
 
         Map<Method, Supplier<InContextInvoker>> handlers = new HashMap<>();
         final Object proxy = Proxy.newProxyInstance(
                 extensionType.getClassLoader(),
-                new Class[] {extensionType},
+                new Class[]{extensionType},
```

Comment:
the style I use would usually keep the space between `[] {`

Reply:
weird.

Reply:
The answer seems to be "IntelliJ has a new checkbox and it was off" :-P